packages/engine/src/database-engines/sqlite.ts

```diff
@@ -48,13 +49,13 @@ export default async (
   });
   const transactions = {
     begin: async (): Promise<void> => {
-      database.run("BEGIN IMMEDIATE TRANSACTION");
+      return database.exec("BEGIN IMMEDIATE TRANSACTION");
```

Comment:
Going to need a transaction ‘stack’ here — keep track to add in the save point — and to know if we have any open transactions so we have an implicit rollback.

Reply:
And — transactions is going to need a read method to know if the stack is balanced on final exit of the entry point, which is really — one transaction left which is the root transaction.